lib/puppet/environments.rb

```diff
@@ -460,7 +460,7 @@ def initialize(value, path)
       def evicted?
         if Time.now > @check_mtime
           new_mtime = File.mtime(path)
-          @check_mtime = Time.now + 10
+          @check_mtime += 10
```

Comment:
If for some reason `evicted?` isn't called for a while, this will mean that `@check_mtime` will fall behind current and mtime-based caching will become ineffective until `@check_mtime` catches up to `Time.now`. I just checked on a Yelp devbox, and our puppet-omnibus ruby Time.now does a syscall-less VDSO read to get the current time. In other words, a call to Time.now costs the same or only slightly more than a local variable read.